Classes/app.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @peoples = []
+  end
+
+  def list_all_books
+    @books.each { |book| puts "Title = #{book.title} Author = #{book.author}" }
+  end
+
+  def list_all_peoples
+    @peoples.each { |people| puts "Name = #{people.name} ID = #{people.id} Age = #{people.age}" }
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Classroom: '
+    classroom = gets.chomp
+    print 'Has parent permission? (Y/N): '
+    parent_permission = gets.chomp.downcase
+    while parent_permission != 'y' && parent_permission != 'n'
+      print 'Has parent permission? (Y/N): '
+      parent_permission = gets.chomp.downcase
+    end
+    student = Student.new(age, classroom, name, parent_permission: true) if parent_permission == 'y'
+    student = Student.new(age, classroom, name, parent_permission: false) if parent_permission == 'n'
+    @peoples << student
+    puts 'Person Created Successfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @peoples << Teacher.new(specialization, age, name)
+    puts 'Person Created Successfully'
+  end
+
+  def create_person
+    puts 'Do you want to create a student(1) or a teacher(2)? [Input the number]: '
+    number = gets.chomp
+    while number != '1' && number != '2'
+      puts 'Do you want to create a student(1) or a teacher(2)? [Input the number 1 or 2 only]: '
+      number = gets.chomp
+    end
+    if number == '1'
+      create_student
+    else
+      create_teacher
+    end
+  end
+
+  def create_book
+    puts 'Title: '
+    title = gets.chomp
+    puts 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index} Title = #{book.title} Author = #{book.author}" }
+    number = gets.chomp.to_i
+    puts 'Select a person from the following list by number(not ID)'
+    @peoples.each.with_index do |person, index|
+      puts "#{index}) #{[person.class.name]} Name #{person.name}, ID #{person.id}, Age #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Date(YYYY-MM-DD)'
+    date = gets.chomp
+    Rental.new(date, @peoples[person_index], @books[number])
+    puts 'Rental created successfully'
+  end
```

Comment:
Nice job fixing the bug from the book creation method👍. Currently, an error is thrown when I try to create a rental and the previous reviewer couldn't point that because a `book` needs to be created before a `rental` can be created🥲. Kindly ensure all features of your app works perfectly before submission so as to avoid any blocker👍.

<img width="530" alt="image" src="https://github.com/abenaTenkorang/school_library/assets/64643995/d9d1cf8f-3db2-402e-995e-34b8990ddafd">

when I try to create a rental with the instructions given on the console, I get a `noMethodError` kindly fix this issue or update the guidelines on the console if there is another way out👍.

Reply:
![Screenshot from 2023-05-19 18-17-37](https://github.com/abenaTenkorang/school_library/assets/109290913/270b5ca6-e0e3-4ba0-9d82-fa0d997e8249)

Bug fixed. Thank you @Ibizugbe